include/ast/instruction.h

```diff
@@ -268,13 +352,34 @@ class Instruction {
 #endif
     // Type 9: IsLast.
     bool IsLast;
+    // Type 10: Try Block
+    struct {
+      uint32_t JumpCatchAll;
+      uint32_t JumpEnd;
+      uint32_t BlockParamNum;
+      uint32_t DelegateIdx;
+      uint32_t VSize;
+      uint32_t HOffset;
+      uint32_t COffset;
+      BlockType ResType;
+    } TryBlock;
+    // Type 11: Tag
+    struct {
+      uint32_t TagIdx;
+    } Tag;
   } Data;
   uint32_t Offset = 0;
   OpCode Code = OpCode::End;
   struct {
     bool IsAllocLabelList : 1;
     bool IsAllocValTypeList : 1;
+    bool IsTryLast : 1;
+    bool IsCatchLast : 1;
+    bool IsDelegate : 1;
   } Flags;
+
+  // TODO: Need refactor
+  std::vector<uint32_t> JumpCatch;
```

Comment:
I would like to ask that is there any better way to deal with this vector? We have to know how many catch clause after loading the entire try-catch-end block, so it can't readily use an uint32_t* inside union to store it.

Reply:
What's the real types used in this vector?

Reply:
Do you mean what value does this vector store? It is used only on try instruction, it stores the index offset on the instruction vector for each catch clause associated with it.

Reply:
One possible way to refactor is to imitate the way to store the lablelist for brtable instruction.

https://github.com/WasmEdge/WasmEdge/blob/9c6d6eef435907e87a64b90e9ae0f8461cb253f3/include/ast/instruction.h#L113-L130

But, we have to go through to the end of the try block to know the size we need for the array to store the index offset (Handling raw pointer is dangerous too). I'm wondering whether there's any better solution.